<!DOCTYPE HTML>
<html>
<head>
<meta charset="UTF-8">
<title>crux cascade-search</title>
<script type="text/javascript"
  src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/javascript">
  MathJax.Hub.Config({jax: ["input/TeX","output/HTML-CSS"], displayAlign: "left"});
</script>
</head>
<body>
<h1>cascade-search</h1>
<h2>Usage:</h2>
<p><code>crux cascade-search [options] &lt;tide spectra file&gt;+ &lt;database-series&gt;</code></p>
<h2>Description:</h2>
<p>Cascade search is a general procedure for incorporating information about peptide groups into the database search and confidence estimation procedure. Peptides may be grouped according to, for example, their enzymatic properties (zero, one, or two enzymatic termini) or the presence of different types of numbers of variable modifications. The algorithm works on a series of databases, each corresponding to a different peptide group. The database is searched in series, and after each search, any spectrum that is identified with a user-specified confidence threshold is sequestered from subsequent searches. The full cascade search procedure is described in this article: </p><blockquote>Attila Kertesz-Farkas, Uri Keich and William Stafford Noble. <a href="http://pubs.acs.org/doi/abs/10.1021/pr501173s">"Tandem mass spectrum identification via cascaded search."</a> <i>Journal of Proteome Research</i>. 14(8):3027-38, 2015. </blockquote>
<h2>Input:</h2>
<ul>
  <li><code>&lt;tide spectra file&gt;+</code> &ndash; The name of one or more files from which to parse the fragmentation spectra, in any of the file formats supported by ProteoWizard. Alternatively, the argument may be one or more binary spectrum files produced by a previous run of crux tide-search using the store-spectra parameter.</li>
  <li><code>&lt;database-series&gt;</code> &ndash; Specify series of databases, each generated by tide-index and separated by comma. Cascade Search will search a set of input spectra against these index files in the given order iteratively.</li>
</ul>
<h2>Output:</h2>
<p>The program writes files to the folder <code>crux-output</code> by default. The name of the output folder can be set by the user using the <code>--output-dir</code> option. The following files will be created:
<ul>
  <li><code>cascade-search.target.txt</code> &ndash; a <a href="txt-format.html">tab-delimited text file</a> containing the target PSMs accepted at a pre-defined fdr.</li>
  <li><code>cascade-search.log.txt</code> &ndash; a log file containing a copy of all messages that were printed to stderr.</li>
  <li><code>cascade-search.params.txt</code> &ndash; a file containing the name and value of all parameters/options for the current operation. Not all parameters in the file may have been used in the operation. The resulting file can be used with the --parameter-file option for other crux programs.</li>
</ul>
<h2>Options:</h2>
<ul style="list-style-type: none;">
<li>
<h3>cascade-search options</h3>
<ul>
  <li><code>--q-value-threshold &lt;float&gt;</code> &ndash; The q-value threshold used by cascade search. Spectra identified with q-value less than this threshold in one search will be excluded from all subsequent searches. Default = <code>0.01</code>.</li>
  <li><code>--estimation-method &lt;mix-max|tdc&gt;</code> &ndash; Specify the method used to estimate q-values: the mix-max procedure or target-decoy competition. Default = <code>tdc</code>.</li>
  <li><code>--score &lt;string&gt;</code> &ndash; Specify the column (for tab-delimited input) or tag (for XML input) used as input to the q-value estimation procedure. If this parameter is unspecified, then assign-confidence tries to seach for "xcorr score", "evalue" (comet), "exact p-value" score fields in this order in the input file. The "smaller-the-better" parameter will then be automatically adjusted. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--smaller-is-better &lt;T|F&gt;</code> &ndash; Specify the semantics of the score, i.e., whether a smaller value implies a better match or vice versa. For example, set this parameter to T for "exact p-value" and F for "xcorr score". Default = <code>false</code>.</li>
  <li><code>--sidak &lt;T|F&gt;</code> &ndash; Adjust the score using the Sidak adjustment and reports them in a new column in the output file. Note that this adjustment only makes sense if the given scores are p-values, and that it requires the presence of the "distinct matches/spectrum" feature for each PSM. Default = <code>false</code>.</li>
  <li><code>--peptide-level &lt;T|F&gt;</code> &ndash; Applied for spectrum-centric search. Eliminates any PSMS for which there exists a better scoring PSM involving the same peptide. This option is incompatible with the mix-max procedure. Default = <code>false</code>.</li>
  <li><code>--combine-charge-states &lt;T|F&gt;</code> &ndash; Specify this parameter to T in order to combine charge states with peptide sequencesin peptide-centric search. Works only if peptide-level=T. Default = <code>false</code>.</li>
  <li><code>--combine-modified-peptides &lt;T|F&gt;</code> &ndash; Specify this parameter to T in order to treat peptides carrying different or no modifications as being the same. Works only if peptide-level=T. Default = <code>false</code>.</li>
</ul>
</li>
<li>
<h3>Search parameters</h3>
<ul>
  <li><code>--precursor-window &lt;float&gt;</code> &ndash; Tolerance used for matching peptides to spectra. Peptides must be within +/- 'precursor-window' of the spectrum value. The precursor window units depend upon precursor-window-type. Default = <code>3</code>.</li>
  <li><code>--precursor-window-type &lt;mass|mz|ppm&gt;</code> &ndash; Specify the units for the window that is used to select peptides around the precursor mass location (mass, mz, ppm). The magnitude of the window is defined by the precursor-window option, and candidate peptides must fall within this window. For the mass window-type, the spectrum precursor m+h value is converted to mass, and the window is defined as that mass +/- precursor-window. If the m+h value is not available, then the mass is calculated from the precursor m/z and provided charge. The peptide mass is computed as the sum of the average amino acid masses plus 18 Da for the terminal OH group. The mz window-type calculates the window as spectrum precursor m/z +/- precursor-window and then converts the resulting m/z range to the peptide mass range using the precursor charge. For the parts-per-million (ppm) window-type, the spectrum mass is calculated as in the mass type. The lower bound of the mass window is then defined as the spectrum mass / (1.0 + (precursor-window / 1000000)) and the upper bound is defined as spectrum mass / (1.0 - (precursor-window / 1000000)). Default = <code>mass</code>.</li>
  <li><code>--spectrum-min-mz &lt;float&gt;</code> &ndash; The lowest spectrum m/z to search in the ms2 file. Default = <code>0</code>.</li>
  <li><code>--spectrum-max-mz &lt;float&gt;</code> &ndash; The highest spectrum m/z to search in the ms2 file. Default = <code>1e+09</code>.</li>
  <li><code>--min-peaks &lt;integer&gt;</code> &ndash; The minimum number of peaks a spectrum must have for it to be searched. Default = <code>20</code>.</li>
  <li><code>--spectrum-charge &lt;1|2|3|all&gt;</code> &ndash; The spectrum charges to search. With 'all' every spectrum will be searched and spectra with multiple charge states will be searched once at each charge state. With 1, 2, or 3 only spectra with that charge state will be searched. Default = <code>all</code>.</li>
  <li><code>--scan-number &lt;string&gt;</code> &ndash; A single scan number or a range of numbers to be searched. Range should be specified as 'first-last' which will include scans 'first' and 'last'. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--compute-sp &lt;T|F&gt;</code> &ndash; Compute the preliminary score Sp for all candidate peptides. Report this score in the output, along with the corresponding rank, the number of matched ions and the total number of ions. This option is recommended if results are to be analyzed by Percolator or Barista. If sqt-output is enabled, then compute-sp is automatically enabled and cannot be overridden. Note that the Sp computation requires re-processing each observed spectrum, so turning on this switch involves significant computational overhead. Default = <code>false</code>.</li>
  <li><code>--remove-precursor-peak &lt;T|F&gt;</code> &ndash; If true, all peaks around the precursor m/z will be removed, within a range specified by the --remove-precursor-tolerance option. Default = <code>false</code>.</li>
  <li><code>--remove-precursor-tolerance &lt;float&gt;</code> &ndash; This parameter specifies the tolerance (in Th) around each precursor m/z that is removed when the --remove-precursor-peak option is invoked. Default = <code>1.5</code>.</li>
  <li><code>--exact-p-value &lt;T|F&gt;</code> &ndash; Enable the calculation of exact p-values for the XCorr score as described in <a href="http://www.ncbi.nlm.nih.gov/pubmed/24895379">this article</a>. Calculation of p-values increases the running time but increases the number of identifications at a fixed confidence threshold. The p-values will be reported in a new column with header "exact p-value", and the "xcorr score" column will be replaced with a "refactored xcorr" column. Note that, currently, p-values can only be computed when the mz-bin-width parameter is set to its default value. Variable and static mods are allowed on non-terminal residues in conjunction with p-value computation, but currently only static mods are allowed on the N-terminus, and no mods on the C-terminus. Default = <code>false</code>.</li>
  <li><code>--use-neutral-loss-peaks &lt;T|F&gt;</code> &ndash; Controls whether neutral loss ions are considered in the search. Two types of neutral losses are included and are applied only to singly charged b- and y-ions: loss of ammonia (NH3, 17.0086343 Da) and H2O (18.0091422). Each neutral loss peak has intensity 1/5 of the primary peak Default = <code>true</code>.</li>
  <li><code>--use-flanking-peaks &lt;T|F&gt;</code> &ndash; Include flanking peaks around singly charged b and y theoretical ions. Each flanking peak occurs in the adjacent m/z bin and has half the intensity of the primary peak. Default = <code>false</code>.</li>
  <li><code>--mz-bin-width &lt;float&gt;</code> &ndash; Before calculation of the XCorr score, the m/z axes of the observed and theoretical spectra are discretized. This parameter specifies the size of each bin. The exact formula is floor((x/mz-bin-width) + 1.0 - mz-bin-offset), where x is the observed m/z value. For low resolution ion trap ms/ms data 1.0005079 and for high resolution ms/ms 0.02 is recommended. Default = <code>1.0005079</code>.</li>
  <li><code>--mz-bin-offset &lt;float&gt;</code> &ndash; In the discretization of the m/z axes of the observed and theoretical spectra, this parameter specifies the location of the left edge of the first bin, relative to mass = 0 (i.e., mz-bin-offset = 0.xx means the left edge of the first bin will be located at +0.xx Da). Default = <code>0.4</code>.</li>
  <li><code>--max-precursor-charge &lt;integer&gt;</code> &ndash; The maximum charge state of a spectra to consider in search. Default = <code>5</code>.</li>
  <li><code>--peptide-centric-search &lt;T|F&gt;</code> &ndash; Carries out a peptide-centric search. For each peptide the top-scoring spectra are reported, in contrast to the standard spectrum-centric search where the top-scoring peptides are reported. Note that in this case the "xcorr rank" column will contain the rank of the given spectrum with respect to the given candidate peptide, rather than vice versa (which is the default). Default = <code>false</code>.</li>
</ul>
</li>
<li>
<h3>Input and output</h3>
<ul>
  <li><code>--decoy-prefix &lt;string&gt;</code> &ndash; Specifies the prefix of the protein names that indicate a decoy. Default = <code>decoy_</code>.</li>
  <li><code>--verbosity &lt;integer&gt;</code> &ndash; Specify the verbosity of the current processes. Each level prints the following messages, including all those at lower verbosity levels: 0-fatal errors, 10-non-fatal errors, 20-warnings, 30-information on the progress of execution, 40-more progress information, 50-debug info, 60-detailed debug info. Default = <code>30</code>.</li>
  <li><code>--parameter-file &lt;string&gt;</code> &ndash; A file containing parameters.  See the <a href="parameter-file.html">parameter documentation</a> page for details. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--overwrite &lt;T|F&gt;</code> &ndash; Replace existing files if true or fail when trying to overwrite a file if false. Default = <code>false</code>.</li>
  <li><code>--output-dir &lt;string&gt;</code> &ndash; The name of the directory where output files will be created. Default = <code>crux-output</code>.</li>
  <li><code>--list-of-files &lt;T|F&gt;</code> &ndash; Specify that the search results are provided as lists of files, rather than as individual files. Default = <code>false</code>.</li>
  <li><code>--fileroot &lt;string&gt;</code> &ndash; The fileroot string will be added as a prefix to all output file names. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--store-spectra &lt;string&gt;</code> &ndash; Specify the name of the file where the binarized fragmentation spectra will be stored. Subsequent runs of crux tide-search will execute more quickly if provided with the spectra in binary format. The filename is specified relative to the current working directory, not the Crux output directory (as specified by --output-dir). This option is not valid if multiple input spectrum files are given. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--store-index &lt;string&gt;</code> &ndash; When providing a FASTA file as the index, the generated binary index will be stored at the given path. This option has no effect if a binary index is provided as the index. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--concat &lt;T|F&gt;</code> &ndash; When set to T, target and decoy search results are reported in a single file, and only the top-scoring N matches (as specified via --top-match) are reported for each spectrum, irrespective of whether the matches involve target or decoy peptides. Default = <code>false</code>.</li>
  <li><code>--print-search-progress &lt;integer&gt;</code> &ndash; Show search progress by printing every n spectra searched. Set to 0 to show no search progress. Default = <code>1000</code>.</li>
  <li><code>--spectrum-parser &lt;pwiz|mstoolkit&gt;</code> &ndash; Specify the parser to use for reading in MS/MS spectra. The default, ProteoWizard parser can read the MS/MS file formats listed <a href="http://proteowizard.sourceforge.net/formats.shtml">here</a>. The alternative is <a href="http://cruxtoolkit.sourceforge.net/mstoolkit.html">MSToolkit parser</a>. If the ProteoWizard parser fails to read your files properly, you may want to try the MSToolkit parser instead. Default = <code>pwiz</code>.</li>
  <li><code>--use-z-line &lt;T|F&gt;</code> &ndash; Specify whether, when parsing an MS2 spectrum file, Crux obtains the precursor mass information from the "S" line or the "Z" line.  Default = <code>true</code>.</li>
  <li><code>--txt-output &lt;T|F&gt;</code> &ndash; Output a tab-delimited results file to the output directory. Default = <code>true</code>.</li>
  <li><code>--sqt-output &lt;T|F&gt;</code> &ndash; Outputs an SQT results file to the output directory. Note that if sqt-output is enabled, then compute-sp is automatically enabled and cannot be overridden. Default = <code>false</code>.</li>
  <li><code>--pepxml-output &lt;T|F&gt;</code> &ndash; Output a pepXML results file to the output directory. Default = <code>false</code>.</li>
  <li><code>--mzid-output &lt;T|F&gt;</code> &ndash; Output an mzIdentML results file to the output directory. Default = <code>false</code>.</li>
  <li><code>--pin-output &lt;T|F&gt;</code> &ndash; Output a Percolator input (PIN) file to the output directory. Default = <code>false</code>.</li>
</ul>
</li>

</ul>
<hr>
<a href="/">Home</a>
</body>
</html>
