<!DOCTYPE HTML>
<html>
<head>
<meta charset="UTF-8">
<title>crux comet</title>
<script type="text/javascript"
  src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/javascript">
  MathJax.Hub.Config({jax: ["input/TeX","output/HTML-CSS"], displayAlign: "left"});
</script>
</head>
<body>
<h1>comet</h1>
<h2>Usage:</h2>
<p><code>crux comet [options] &lt;input spectra&gt;+ &lt;database name&gt;</code></p>
<h2>Description:</h2>
<p>This command searches a protein database with a set of spectra, assigning peptide sequences to the observed spectra. This search engine was developed by Jimmy Eng at the University of Washington Proteomics Resource.</p><p>Although its history goes back two decades, the Comet search engine was first made publicly available in August 2012 on SourceForge. Comet is multithreaded and supports multiple input and output formats.</p><blockquote><a href="http://onlinelibrary.wiley.com/doi/10.1002/pmic.201200439/abstract">&quot;Comet: an open source tandem mass spectrometry sequence database search tool.&quot;</a> Eng JK, Jahan TA, Hoopmann MR. <em>Proteomics</em>. 2012 Nov 12. doi: 10.1002/pmic201200439</blockquote>
<h2>Input:</h2>
<ul>
  <li><code>&lt;input spectra&gt;+</code> &ndash; The name of the file from which to parse the spectra. Valid formats include mzXML, mzML, mz5, raw, ms2, and cms2. Files in mzML or mzXML may be compressed with gzip. RAW files can be parsed only under windows and if the appropriate libraries were included at compile time.</li>
  <li><code>&lt;database name&gt;</code> &ndash; A full or relative path to the sequence database, in FASTA format, to search. Example databases include RefSeq or UniProt.  The database can contain amino acid sequences or nucleic acid sequences. If sequences are amino acid sequences, set the parameter "nucleotide_reading_frame = 0". If the sequences are nucleic acid sequences, you must instruct Comet to translate these to amino acid sequences. Do this by setting nucleotide_reading_frame" to a value between 1 and 9.</li>
</ul>
<h2>Output:</h2>
<p>The program writes files to the folder <code>crux-output</code> by default. The name of the output folder can be set by the user using the <code>--output-dir</code> option. The following files will be created:
<ul>
  <li><code>comet.target.txt</code> &ndash; a tab-delimited text file containing the target PSMs. See <a href="txt-format.html">txt file format</a> for a list of the fields.</li>
  <li><code>comet.params.txt</code> &ndash; a file containing the name and value of all parameters/options for the current operation. Not all parameters in the file may have been used in the operation. The resulting file can be used with the --parameter-file option for other crux programs.</li>
  <li><code>comet.log.txt</code> &ndash; a log file containing a copy of all messages that were printed to standard error.</li>
</ul>
<h2>Options:</h2>
<ul style="list-style-type: none;">
<li>
<h3>Database</h3>
<ul>
  <li><code>--decoy_search &lt;integer&gt;</code> &ndash; 0=no, 1=concatenated search, 2=separate search. Default = <code>0</code>.</li>
</ul>
</li>
<li>
<h3>CPU threads</h3>
<ul>
  <li><code>--num_threads &lt;integer&gt;</code> &ndash; 0=poll CPU to set num threads; else specify num threads directly (max 32). Default = <code>0</code>.</li>
</ul>
</li>
<li>
<h3>Masses</h3>
<ul>
  <li><code>--peptide_mass_tolerance &lt;float&gt;</code> &ndash; Controls the mass tolerance value.  The mass tolerance is set at +/- the specified number i.e. an entered value of "1.0" applies a -1.0 to +1.0 tolerance. The units of the mass tolerance is controlled by the parameter "peptide_mass_units".  Default = <code>3</code>.</li>
  <li><code>--peptide_mass_units &lt;integer&gt;</code> &ndash; 0=amu, 1=mmu, 2=ppm. Default = <code>0</code>.</li>
  <li><code>--mass_type_parent &lt;integer&gt;</code> &ndash; 0=average masses, 1=monoisotopic masses. Default = <code>1</code>.</li>
  <li><code>--mass_type_fragment &lt;integer&gt;</code> &ndash; 0=average masses, 1=monoisotopic masses. Default = <code>1</code>.</li>
  <li><code>--isotope_error &lt;integer&gt;</code> &ndash; 0=off, 1=on -1/0/1/2/3 (standard C13 error), 2=-8/-4/0/4/8 (for +4/+8 labeling). Default = <code>0</code>.</li>
</ul>
</li>
<li>
<h3>Search enzyme</h3>
<ul>
  <li><code>--search_enzyme_number &lt;integer&gt;</code> &ndash; Specify a search enzyme from the end of the parameter file. Default = <code>1</code>.</li>
  <li><code>--num_enzyme_termini &lt;integer&gt;</code> &ndash; valid values are 1 (semi-digested), 2 (fully digested), 8 N-term, 9 C-term. Default = <code>2</code>.</li>
  <li><code>--allowed_missed_cleavage &lt;integer&gt;</code> &ndash; maximum value is 5; for enzyme search Default = <code>2</code>.</li>
</ul>
</li>
<li>
<h3>Fragment ions</h3>
<ul>
  <li><code>--fragment_bin_tol &lt;float&gt;</code> &ndash; binning to use on fragment ions. Default = <code>1.000507</code>.</li>
  <li><code>--fragment_bin_offset &lt;float&gt;</code> &ndash; offset position to start the binning (0.0 to 1.0). Default = <code>0.4</code>.</li>
  <li><code>--theoretical_fragment_ions &lt;integer&gt;</code> &ndash; 0=default peak shape, 1=M peak only. Default = <code>1</code>.</li>
  <li><code>--use_A_ions &lt;integer&gt;</code> &ndash; Controls whether or not A-ions are considered in the search (0 - no, 1 - yes). Default = <code>0</code>.</li>
  <li><code>--use_B_ions &lt;integer&gt;</code> &ndash; Controls whether or not B-ions are considered in the search (0 - no, 1 - yes). Default = <code>1</code>.</li>
  <li><code>--use_C_ions &lt;integer&gt;</code> &ndash; Controls whether or not C-ions are considered in the search (0 - no, 1 - yes). Default = <code>0</code>.</li>
  <li><code>--use_X_ions &lt;integer&gt;</code> &ndash; Controls whether or not X-ions are considered in the search (0 - no, 1 - yes). Default = <code>0</code>.</li>
  <li><code>--use_Y_ions &lt;integer&gt;</code> &ndash; Controls whether or not Y-ions are considered in the search (0 - no, 1 - yes). Default = <code>1</code>.</li>
  <li><code>--use_Z_ions &lt;integer&gt;</code> &ndash; Controls whether or not Z-ions are considered in the search (0 - no, 1 - yes). Default = <code>0</code>.</li>
  <li><code>--use_NL_ions &lt;integer&gt;</code> &ndash; 0=no, 1= yes to consider NH3/H2O neutral loss peak. Default = <code>1</code>.</li>
  <li><code>--use_sparse_matrix &lt;integer&gt;</code> &ndash; Controls whether or not internal sparse matrix data representation is used. Default = <code>0</code>.</li>
</ul>
</li>
<li>
<h3>mzXML/mzML parameters</h3>
<ul>
  <li><code>--scan_range &lt;string&gt;</code> &ndash; start and scan scan range to search; 0 as 1st entry ignores parameter. Default = <code>0 0</code>.</li>
  <li><code>--precursor_charge &lt;string&gt;</code> &ndash; precursor charge range to analyze; does not override mzXML charge; 0 as 1st entry ignores parameter. Default = <code>0 0</code>.</li>
  <li><code>--ms_level &lt;integer&gt;</code> &ndash; MS level to analyze, valid are levels 2 or 3. Default = <code>2</code>.</li>
  <li><code>--activation_method &lt;ALL|CID|ECD|ETD|PQD|HCD|IRMPD&gt;</code> &ndash; Specifies which scan types are searched. Default = <code>ALL</code>.</li>
  <li><code>--override_charge &lt;integer&gt;</code> &ndash; specifies the whether to override existing precursor charge state information when present in the files with the charge range specified by the "precursor_charge" parameter. Default = <code>0</code>.</li>
</ul>
</li>
<li>
<h3>Misc. parameters</h3>
<ul>
  <li><code>--output_suffix &lt;string&gt;</code> &ndash; specifies the suffix string that is appended to the base output name for the pep.xml, pin.xml, txt and sqt output files. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--digest_mass_range &lt;string&gt;</code> &ndash; MH+ peptide mass range to analyze. Default = <code>600.0 5000.0</code>.</li>
  <li><code>--num_results &lt;integer&gt;</code> &ndash; number of search hits to store internally. Default = <code>50</code>.</li>
  <li><code>--skip_researching &lt;integer&gt;</code> &ndash; for '.out' file output only, 0=search everything again, 1=don't search if .out exists. Default = <code>1</code>.</li>
  <li><code>--max_fragment_charge &lt;integer&gt;</code> &ndash; set maximum fragment charge state to analyze (allowed max 5). Default = <code>3</code>.</li>
  <li><code>--max_precursor_charge &lt;integer&gt;</code> &ndash; set maximum precursor charge state to analyze (allowed max 9). Default = <code>6</code>.</li>
  <li><code>--nucleotide_reading_frame &lt;integer&gt;</code> &ndash; 0=proteinDB, 1-6, 7=forward three, 8=reverse three, 9=all six. Default = <code>0</code>.</li>
  <li><code>--clip_nterm_methionine &lt;integer&gt;</code> &ndash; 0=leave sequences as-is; 1=also consider sequence w/o N-term methionine. Default = <code>0</code>.</li>
  <li><code>--spectrum_batch_size &lt;integer&gt;</code> &ndash; max. # of spectra to search at a time; 0 to search the entire scan range in one loop. Default = <code>0</code>.</li>
</ul>
</li>
<li>
<h3>Spectral processing</h3>
<ul>
  <li><code>--minimum_peaks &lt;integer&gt;</code> &ndash; minimum num. of peaks in spectrum to search. Default = <code>10</code>.</li>
  <li><code>--minimum_intensity &lt;float&gt;</code> &ndash; minimum intensity value to read in. Default = <code>0</code>.</li>
  <li><code>--remove_precursor_peak &lt;integer&gt;</code> &ndash; 0=no, 1=yes, 2=all charge reduced precursor peaks (for ETD). Default = <code>0</code>.</li>
  <li><code>--remove_precursor_tolerance &lt;float&gt;</code> &ndash; +- Da tolerance for precursor removal. Default = <code>1.5</code>.</li>
  <li><code>--clear_mz_range &lt;string&gt;</code> &ndash; for iTRAQ/TMT type data; will clear out all peaks in the specified m/z range. Default = <code>0.0 0.0</code>.</li>
</ul>
</li>
<li>
<h3>Variable modifications</h3>
<ul>
  <li><code>--variable_mod01 &lt;string&gt;</code> &ndash; Up to 9 variable modifications are supported; format: "<mass> <residues> <0=variable/1=binary> <max mods per a peptide>" e.g. 79.966331 STY 0 3. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--variable_mod02 &lt;string&gt;</code> &ndash; Up to 9 variable modifications are supported; format: "<mass> <residues> <0=variable/1=binary> <max mods per a peptide>" e.g. 79.966331 STY 0 3. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--variable_mod03 &lt;string&gt;</code> &ndash; Up to 9 variable modifications are supported; format: "<mass> <residues> <0=variable/1=binary> <max mods per a peptide>" e.g. 79.966331 STY 0 3. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--variable_mod04 &lt;string&gt;</code> &ndash; Up to 9 variable modifications are supported; format: "<mass> <residues> <0=variable/1=binary> <max mods per a peptide>" e.g. 79.966331 STY 0 3. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--variable_mod05 &lt;string&gt;</code> &ndash; Up to 9 variable modifications are supported; format: "<mass> <residues> <0=variable/1=binary> <max mods per a peptide>" e.g. 79.966331 STY 0 3. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--variable_mod06 &lt;string&gt;</code> &ndash; Up to 9 variable modifications are supported; format: "<mass> <residues> <0=variable/1=binary> <max mods per a peptide>" e.g. 79.966331 STY 0 3. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--variable_mod07 &lt;string&gt;</code> &ndash; Up to 9 variable modifications are supported; format: "<mass> <residues> <0=variable/1=binary> <max mods per a peptide>" e.g. 79.966331 STY 0 3. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--variable_mod08 &lt;string&gt;</code> &ndash; Up to 9 variable modifications are supported; format: "<mass> <residues> <0=variable/1=binary> <max mods per a peptide>" e.g. 79.966331 STY 0 3. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--variable_mod09 &lt;string&gt;</code> &ndash; Up to 9 variable modifications are supported; format: "<mass> <residues> <0=variable/1=binary> <max mods per a peptide>" e.g. 79.966331 STY 0 3. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--require_variable_mod &lt;integer&gt;</code> &ndash; controls whether the analyzed peptides must contain at least one variable modification. Default = <code>0</code>.</li>
  <li><code>--max_variable_mods_in_peptide &lt;integer&gt;</code> &ndash; Specifies the total/maximum number of residues that can be modified in a peptide. Default = <code>5</code>.</li>
</ul>
</li>
<li>
<h3>Static modifications</h3>
<ul>
  <li><code>--add_Cterm_peptide &lt;float&gt;</code> &ndash; Specifiy a static modification to the c-terminus of all peptides. Default = <code>0</code>.</li>
  <li><code>--add_Nterm_peptide &lt;float&gt;</code> &ndash; Specify a static modification to the n-terminus of all peptides. Default = <code>0</code>.</li>
  <li><code>--add_Cterm_protein &lt;float&gt;</code> &ndash; Specify a static modification to the c-terminal peptide of each protein. Default = <code>0</code>.</li>
  <li><code>--add_Nterm_protein &lt;float&gt;</code> &ndash; Specify a static modification to the n-terminal peptide of each protein. Default = <code>0</code>.</li>
  <li><code>--add_A_alanine &lt;float&gt;</code> &ndash; Specify a static modification to the residue A. Default = <code>0</code>.</li>
  <li><code>--add_B_user_amino_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue B. Default = <code>0</code>.</li>
  <li><code>--add_C_cysteine &lt;float&gt;</code> &ndash; Specify a static modification to the residue C. Default = <code>57.021464</code>.</li>
  <li><code>--add_D_aspartic_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue D. Default = <code>0</code>.</li>
  <li><code>--add_E_glutamic_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue E. Default = <code>0</code>.</li>
  <li><code>--add_F_phenylalanine &lt;float&gt;</code> &ndash; Specify a static modification to the residue F. Default = <code>0</code>.</li>
  <li><code>--add_G_glycine &lt;float&gt;</code> &ndash; Specify a static modification to the residue G. Default = <code>0</code>.</li>
  <li><code>--add_H_histidine &lt;float&gt;</code> &ndash; Specify a static modification to the residue H. Default = <code>0</code>.</li>
  <li><code>--add_I_isoleucine &lt;float&gt;</code> &ndash; Specify a static modification to the residue I. Default = <code>0</code>.</li>
  <li><code>--add_J_user_amino_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue J. Default = <code>0</code>.</li>
  <li><code>--add_K_lysine &lt;float&gt;</code> &ndash; Specify a static modification to the residue K. Default = <code>0</code>.</li>
  <li><code>--add_L_leucine &lt;float&gt;</code> &ndash; Specify a static modification to the residue L. Default = <code>0</code>.</li>
  <li><code>--add_M_methionine &lt;float&gt;</code> &ndash; Specify a static modification to the residue M. Default = <code>0</code>.</li>
  <li><code>--add_N_asparagine &lt;float&gt;</code> &ndash; Specify a static modification to the residue N. Default = <code>0</code>.</li>
  <li><code>--add_O_ornithine &lt;float&gt;</code> &ndash; Specify a static modification to the residue O. Default = <code>0</code>.</li>
  <li><code>--add_P_proline &lt;float&gt;</code> &ndash; Specify a static modification to the residue P. Default = <code>0</code>.</li>
  <li><code>--add_Q_glutamine &lt;float&gt;</code> &ndash; Specify a static modification to the residue Q. Default = <code>0</code>.</li>
  <li><code>--add_R_arginine &lt;float&gt;</code> &ndash; Specify a static modification to the residue R. Default = <code>0</code>.</li>
  <li><code>--add_S_serine &lt;float&gt;</code> &ndash; Specify a static modification to the residue S. Default = <code>0</code>.</li>
  <li><code>--add_T_threonine &lt;float&gt;</code> &ndash; Specify a static modification to the residue T. Default = <code>0</code>.</li>
  <li><code>--add_U_user_amino_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue U. Default = <code>0</code>.</li>
  <li><code>--add_V_valine &lt;float&gt;</code> &ndash; Specify a static modification to the residue V. Default = <code>0</code>.</li>
  <li><code>--add_W_tryptophan &lt;float&gt;</code> &ndash; Specify a static modification to the residue W. Default = <code>0</code>.</li>
  <li><code>--add_X_user_amino_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue X. Default = <code>0</code>.</li>
  <li><code>--add_Y_tyrosine &lt;float&gt;</code> &ndash; Specify a static modification to the residue Y. Default = <code>0</code>.</li>
  <li><code>--add_Z_user_amino_acid &lt;float&gt;</code> &ndash; Specify a static modification to the residue Z. Default = <code>0</code>.</li>
</ul>
</li>
<li>
<h3>Input and output</h3>
<ul>
  <li><code>--fileroot &lt;string&gt;</code> &ndash; The fileroot string will be added as a prefix to all output file names. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--output-dir &lt;string&gt;</code> &ndash; The name of the directory where output files will be created. Default = <code>crux-output</code>.</li>
  <li><code>--overwrite &lt;T|F&gt;</code> &ndash; Replace existing files if true or fail when trying to overwrite a file if false. Default = <code>false</code>.</li>
  <li><code>--parameter-file &lt;string&gt;</code> &ndash; A file containing parameters.  See the <a href="parameter-file.html">parameter documentation</a> page for details. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--verbosity &lt;integer&gt;</code> &ndash; Specify the verbosity of the current processes. Each level prints the following messages, including all those at lower verbosity levels: 0-fatal errors, 10-non-fatal errors, 20-warnings, 30-information on the progress of execution, 40-more progress information, 50-debug info, 60-detailed debug info. Default = <code>30</code>.</li>
  <li><code>--output_sqtfile &lt;integer&gt;</code> &ndash; 0=no, 1=yes  write sqt file. Default = <code>0</code>.</li>
  <li><code>--output_pepxmlfile &lt;integer&gt;</code> &ndash; 0=no, 1=yes  write pep.xml file. Default = <code>1</code>.</li>
  <li><code>--output_percolatorfile &lt;integer&gt;</code> &ndash; 0=no, 1=yes write percolator file. Default = <code>0</code>.</li>
  <li><code>--output_txtfile &lt;integer&gt;</code> &ndash; 0=no, 1=yes  write tab-delimited text file. Default = <code>1</code>.</li>
  <li><code>--output_outfiles &lt;integer&gt;</code> &ndash; 0=no, 1=yes  write .out files. Default = <code>0</code>.</li>
  <li><code>--print_expect_score &lt;integer&gt;</code> &ndash; 0=no, 1=yes to replace Sp with expect in out & sqt. Default = <code>1</code>.</li>
  <li><code>--num_output_lines &lt;integer&gt;</code> &ndash; num peptide results to show. Default = <code>5</code>.</li>
  <li><code>--show_fragment_ions &lt;integer&gt;</code> &ndash; 0=no, 1=yes for out files only. Default = <code>0</code>.</li>
  <li><code>--sample_enzyme_number &lt;integer&gt;</code> &ndash; Sample enzyme which is possibly different than the one applied to the search. Used to calculate NTT & NMC in pepXML output. Default = <code>1</code>.</li>
</ul>
</li>

</ul>
<hr>
<a href="/">Home</a>
</body>
</html>
