<!DOCTYPE HTML>
<html>
<head>
<meta charset="UTF-8">
<title>crux search-for-xlinks</title>
<script type="text/javascript"
  src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/javascript">
  MathJax.Hub.Config({jax: ["input/TeX","output/HTML-CSS"], displayAlign: "left"});
</script>
</head>
<body>
<h1>search-for-xlinks</h1>
<h2>Usage:</h2>
<p><code>crux search-for-xlinks [options] &lt;ms2 file&gt; &lt;protein fasta file&gt; &lt;link sites&gt; &lt;link mass&gt;</code></p>
<h2>Description:</h2>
<p>This command compares a set of spectra to cross-linked peptides derived from a protein database in FASTA format. For each spectrum, the program generates a list of candidate molecules, including linear peptides, dead-end products, self-loop products and cross-linked products, with masses that lie within a specified range of the spectrum's precursor mass. These candidate molecules are ranked using XCorr, and the XCorr scores are assigned statistical confidence estimates using an empirical curve fitting procedure.</p><p>The algorithm is described in more detail in the following article:</p><blockquote>Sean McIlwain, Paul Draghicescu, Pragya Singh, David R. Goodlett and William Stafford Noble. <a href="http://pubs.acs.org/doi/abs/10.1021/pr901163d">&quot;Detecting cross-linked peptides by searching against a database of cross-linked peptide pairs.&quot;</a> <em>Journal of Proteome Research</em>. 2010.</blockquote>
<h2>Input:</h2>
<ul>
  <li><code>&lt;ms2 file&gt;</code> &ndash; File containing spectra to be searched.</li>
  <li><code>&lt;protein fasta file&gt;</code> &ndash; The name of the file in FASTA format from which to retrieve proteins.</li>
  <li><code>&lt;link sites&gt;</code> &ndash; A comma delimited list of the amino acids to allow cross-links with. For example, "A:K,A:D" means that the cross linker can attach A to K or A to D. Cross-links involving the N-terminus of a protein can be specified as a link site by using "nterm". For example, "nterm:K" means that a cross-link can attach a protein's N-terminus to a lysine.</li>
  <li><code>&lt;link mass&gt;</code> &ndash; The mass modification of the linker when attached to a peptide.</li>
</ul>
<h2>Output:</h2>
<p>The program writes files to the folder <code>crux-output</code> by default. The name of the output folder can be set by the user using the <code>--output-dir</code> option. The following files will be created:
<ul>
  <li><code>search-for-xlinks.target.txt</code> &ndash; a tab-delimited text file containing the peptide-spectrum matches (PSMs). See the <a href="txt-format.html">txt file format</a> for a list of the fields.</li>
  <li><code>search-for-xlinks.decoy.txt</code> &ndash; a tab-delimited text file containing the decoy PSMs. See the <a href="txt-format.html">txt file format</a> for a list of the fields.</li>
  <li><code>search-for-xlinks.qvalues.txt</code> &ndash; a tab-delimited text file containing the top ranked PSMs with calculated q-values. See the <a href="txt-format.html">txt file format</a> for a list of the fields.</li>
  <li><code>search-for-xlinks.params.txt</code> &ndash; a file containing the name and value of all parameters/options for the current operation. Not all parameters in the file may have been used in the operation. The resulting file can be used with the --parameter-file option for other crux programs.</li>
  <li><code>search-for-xlinks.log.txt</code> &ndash; a log file containing a copy of all messages that were printed to stderr.</li>
</ul>
<h2>Options:</h2>
<ul style="list-style-type: none;">
<li>
<h3>Peptide properties</h3>
<ul>
  <li><code>--min-mass &lt;float&gt;</code> &ndash; The minimum mass (in Da) of peptides to consider. Default = <code>200</code>.</li>
  <li><code>--max-mass &lt;float&gt;</code> &ndash; The maximum mass (in Da) of peptides to consider. Default = <code>7200</code>.</li>
  <li><code>--min-length &lt;integer&gt;</code> &ndash; The minimum length of peptides to consider. Default = <code>6</code>.</li>
  <li><code>--max-length &lt;integer&gt;</code> &ndash; The maximum length of peptides to consider. Default = <code>50</code>.</li>
  <li><code>--isotopic-mass &lt;average|mono&gt;</code> &ndash; Specify the type of isotopic masses to use when calculating the peptide mass. Default = <code>average</code>.</li>
</ul>
</li>
<li>
<h3>Amino acid modifications</h3>
<ul>
  <li><code>--mod &lt;string&gt;</code> &ndash; &lt;mass change&gt;:&lt;aa list&gt;:&lt;max per peptide&gt;:&lt;prevents cleavage&gt;:&lt;prevents cross-link&gt;. Consider modifications on any amino acid in aa list with at most max-per-peptide in one peptide. This parameter may be included with different values multiple times so long as the total number of mod, cmod, and nmod parameters does not exceed 11. The prevents cleavage and prevents cross-link are T/F optional arguments for describing whether the modification prevents enzymatic cleavage of cross-linking respectively. This option is only available when use-old-xlink=F. Default = <code>NO MODS</code>.</li>
  <li><code>--max-mods &lt;integer&gt;</code> &ndash; The maximum number of modifications that can be applied to a single peptide. Default = <code>255</code>.</li>
</ul>
</li>
<li>
<h3>Decoy database generation</h3>
<ul>
  <li><code>--seed &lt;string&gt;</code> &ndash; When given a unsigned integer value seeds the random number generator with that value. When given the string "time" seeds the random number generator with the system time. Default = <code>1</code>.</li>
</ul>
</li>
<li>
<h3>Enzymatic digestion</h3>
<ul>
  <li><code>--enzyme &lt;no-enzyme|trypsin|trypsin/p|chymotrypsin|elastase|clostripain|cyanogen-bromide|iodosobenzoate|proline-endopeptidase|staph-protease|asp-n|lys-c|lys-n|arg-c|glu-c|pepsin-a|elastase-trypsin-chymotrypsin|custom-enzyme&gt;</code> &ndash; Specify the enzyme used to digest the proteins in silico. Available enzymes (with the corresponding digestion rules indicated in parentheses) include no-enzyme ([X]|[X]), trypsin ([RK]|{P}), trypsin/p ([RK]|[]), chymotrypsin ([FWYL]|{P}), elastase ([ALIV]|{P}), clostripain ([R]|[]), cyanogen-bromide ([M]|[]), iodosobenzoate ([W]|[]), proline-endopeptidase ([P]|[]), staph-protease ([E]|[]), asp-n ([]|[D]), lys-c ([K]|{P}), lys-n ([]|[K]), arg-c ([R]|{P}), glu-c ([DE]|{P}), pepsin-a ([FL]|{P}), elastase-trypsin-chymotrypsin ([ALIVKRWFY]|{P}). Specifying --enzyme no-enzyme yields a non-enzymatic digest. <strong>Warning:</strong> the resulting index may be quite large. Default = <code>trypsin</code>.</li>
  <li><code>--custom-enzyme &lt;string&gt;</code> &ndash; Specify rules for in silico digestion of protein sequences. Overrides the enzyme option. Two lists of residues are given enclosed in square brackets or curly braces and separated by a |. The first list contains residues required/prohibited before the cleavage site and the second list is residues after the cleavage site. If the residues are required for digestion, they are in square brackets, '[' and ']'. If the residues prevent digestion, then they are enclosed in curly braces, '{' and '}'. Use X to indicate all residues. For example, trypsin cuts after R or K but not before P which is represented as [RK]|{P}. AspN cuts after any residue but only before D which is represented as [X]|[D]. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--digestion &lt;full-digest|partial-digest|non-specific-digest&gt;</code> &ndash; Specify whether every peptide in the database must have two enzymatic termini (full-digest) or if peptides with only one enzymatic terminus are also included (partial-digest). Default = <code>full-digest</code>.</li>
  <li><code>--missed-cleavages &lt;integer&gt;</code> &ndash; Maximum number of missed cleavages per peptide to allow in enzymatic digestion. Default = <code>0</code>.</li>
</ul>
</li>
<li>
<h3>Search parameters</h3>
<ul>
  <li><code>--spectrum-min-mz &lt;float&gt;</code> &ndash; The lowest spectrum m/z to search in the ms2 file. Default = <code>0</code>.</li>
  <li><code>--spectrum-max-mz &lt;float&gt;</code> &ndash; The highest spectrum m/z to search in the ms2 file. Default = <code>1e+09</code>.</li>
  <li><code>--spectrum-charge &lt;1|2|3|all&gt;</code> &ndash; The spectrum charges to search. With 'all' every spectrum will be searched and spectra with multiple charge states will be searched once at each charge state. With 1, 2, or 3 only spectra with that charge state will be searched. Default = <code>all</code>.</li>
  <li><code>--compute-sp &lt;T|F&gt;</code> &ndash; Compute the preliminary score Sp for all candidate peptides. Report this score in the output, along with the corresponding rank, the number of matched ions and the total number of ions. This option is recommended if results are to be analyzed by Percolator or Barista. If sqt-output is enabled, then compute-sp is automatically enabled and cannot be overridden. Note that the Sp computation requires re-processing each observed spectrum, so turning on this switch involves significant computational overhead. Default = <code>false</code>.</li>
  <li><code>--precursor-window &lt;float&gt;</code> &ndash; Tolerance used for matching peptides to spectra. Peptides must be within +/- 'precursor-window' of the spectrum value. The precursor window units depend upon precursor-window-type. Default = <code>3</code>.</li>
  <li><code>--precursor-window-type &lt;mass|mz|ppm&gt;</code> &ndash; Specify the units for the window that is used to select peptides around the precursor mass location (mass, mz, ppm). The magnitude of the window is defined by the precursor-window option, and candidate peptides must fall within this window. For the mass window-type, the spectrum precursor m+h value is converted to mass, and the window is defined as that mass +/- precursor-window. If the m+h value is not available, then the mass is calculated from the precursor m/z and provided charge. The peptide mass is computed as the sum of the average amino acid masses plus 18 Da for the terminal OH group. The mz window-type calculates the window as spectrum precursor m/z +/- precursor-window and then converts the resulting m/z range to the peptide mass range using the precursor charge. For the parts-per-million (ppm) window-type, the spectrum mass is calculated as in the mass type. The lower bound of the mass window is then defined as the spectrum mass / (1.0 + (precursor-window / 1000000)) and the upper bound is defined as spectrum mass / (1.0 - (precursor-window / 1000000)). Default = <code>mass</code>.</li>
  <li><code>--precursor-window-weibull &lt;float&gt;</code> &ndash; Search decoy peptides within +/- precursor-window-weibull of the precursor mass. The resulting scores are used only for fitting the Weibull distribution Default = <code>20</code>.</li>
  <li><code>--min-weibull-points &lt;integer&gt;</code> &ndash; Keep shuffling and collecting XCorr scores until the minimum number of points for weibull fitting (using targets and decoys) is achieved. Default = <code>4000</code>.</li>
  <li><code>--max-ion-charge &lt;string&gt;</code> &ndash; Predict theoretical ions up to max charge state (1, 2, ... ,6) or up to the charge state of the peptide ("peptide"). If the max-ion-charge is greater than the charge state of the peptide, then the maximum is the peptide charge.  Default = <code>peptide</code>.</li>
  <li><code>--scan-number &lt;string&gt;</code> &ndash; A single scan number or a range of numbers to be searched. Range should be specified as 'first-last' which will include scans 'first' and 'last'. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--mz-bin-width &lt;float&gt;</code> &ndash; Before calculation of the XCorr score, the m/z axes of the observed and theoretical spectra are discretized. This parameter specifies the size of each bin. The exact formula is floor((x/mz-bin-width) + 1.0 - mz-bin-offset), where x is the observed m/z value. For low resolution ion trap ms/ms data 1.0005079 and for high resolution ms/ms 0.02 is recommended. Default = <code>1.0005079</code>.</li>
  <li><code>--mz-bin-offset &lt;float&gt;</code> &ndash; In the discretization of the m/z axes of the observed and theoretical spectra, this parameter specifies the location of the left edge of the first bin, relative to mass = 0 (i.e., mz-bin-offset = 0.xx means the left edge of the first bin will be located at +0.xx Da). Default = <code>0.4</code>.</li>
  <li><code>--mod-mass-format &lt;mod-only|total|separate&gt;</code> &ndash; Specify how sequence modifications are reported in various output files. Each modification is reported as a number enclosed in square braces following the modified residue; however the number may correspond to one of three different masses: (1) 'mod-only' reports the value of the mass shift induced by the modification; (2) 'total' reports the mass of the residue with the modification (residue mass plus modification mass); (3) 'separate' is the same as 'mod-only', but multiple modifications to a single amino acid are reported as a comma-separated list of values. For example, suppose amino acid D has an unmodified mass of 115 as well as two moifications of masses +14 and +2. In this case, the amino acid would be reported as D[16] with 'mod-only', D[131] with 'total', and D[14,2] with 'separate'. Default = <code>mod-only</code>.</li>
  <li><code>--use-flanking-peaks &lt;T|F&gt;</code> &ndash; Include flanking peaks around singly charged b and y theoretical ions. Each flanking peak occurs in the adjacent m/z bin and has half the intensity of the primary peak. Default = <code>false</code>.</li>
  <li><code>--fragment-mass &lt;average|mono&gt;</code> &ndash; Specify which isotopes to use in calculating fragment ion mass. Default = <code>mono</code>.</li>
  <li><code>--isotope-windows &lt;string&gt;</code> &ndash; Provides a list of isotopic windows to search. For example, -1,0,1 will search in three disjoint windows: (1) precursor_mass - neutron_mass +/- window, (2) precursor_mass +/- window, and (3) precursor_mass + neutron_mass +/- window. The window size is defined from the precursor-window and precursor-window-type parameters. This option is only available when use-old-xlink=F. Default = <code>0</code>.</li>
  <li><code>--compute-p-values &lt;T|F&gt;</code> &ndash; Estimate the parameters of the score distribution for each spectrum by fitting to a Weibull distribution, and compute a p-value for each xlink product. This option is only available when use-old-xlink=F. Default = <code>false</code>.</li>
</ul>
</li>
<li>
<h3>Cross-linking parameters</h3>
<ul>
  <li><code>--use-old-xlink &lt;T|F&gt;</code> &ndash; Use the old version of xlink-searching algorithm. When false, a new version of the code is run. The new version supports variable modifications and can handle more complex databases. This new code is still in development and should be considered a beta release. Default = <code>true</code>.</li>
  <li><code>--xlink-include-linears &lt;T|F&gt;</code> &ndash; Include linear peptides in the search. Default = <code>true</code>.</li>
  <li><code>--xlink-include-deadends &lt;T|F&gt;</code> &ndash; Include dead-end peptides in the search. Default = <code>true</code>.</li>
  <li><code>--xlink-include-selfloops &lt;T|F&gt;</code> &ndash; Include self-loop peptides in the search. Default = <code>true</code>.</li>
  <li><code>--xlink-include-inter &lt;T|F&gt;</code> &ndash; Include inter-protein cross-link candidates within the search. Default = <code>true</code>.</li>
  <li><code>--xlink-include-intra &lt;T|F&gt;</code> &ndash; Include intra-protein cross-link candiates within the search. Default = <code>true</code>.</li>
  <li><code>--xlink-include-inter-intra &lt;T|F&gt;</code> &ndash; Include crosslink candidates that are both inter and intra. Default = <code>true</code>.</li>
  <li><code>--xlink-prevents-cleavage &lt;string&gt;</code> &ndash; List of amino acids for which the cross-linker can prevent cleavage. This option is only available when use-old-xlink=F. Default = <code>K</code>.</li>
  <li><code>--max-xlink-mods &lt;integer&gt;</code> &ndash; Specify the maximum number of modifications allowed on a crosslinked peptide. This option is only available when use-old-xlink=F. Default = <code>0</code>.</li>
</ul>
</li>
<li>
<h3>Input and output</h3>
<ul>
  <li><code>--spectrum-parser &lt;pwiz|mstoolkit&gt;</code> &ndash; Specify the parser to use for reading in MS/MS spectra. The default, ProteoWizard parser can read the MS/MS file formats listed <a href="http://proteowizard.sourceforge.net/formats.shtml">here</a>. The alternative is <a href="http://cruxtoolkit.sourceforge.net/mstoolkit.html">MSToolkit parser</a>. If the ProteoWizard parser fails to read your files properly, you may want to try the MSToolkit parser instead. Default = <code>pwiz</code>.</li>
  <li><code>--use-z-line &lt;T|F&gt;</code> &ndash; Specify whether, when parsing an MS2 spectrum file, Crux obtains the precursor mass information from the "S" line or the "Z" line.  Default = <code>true</code>.</li>
  <li><code>--top-match &lt;integer&gt;</code> &ndash; Specify the number of matches to report for each spectrum. Default = <code>5</code>.</li>
  <li><code>--output-dir &lt;string&gt;</code> &ndash; The name of the directory where output files will be created. Default = <code>crux-output</code>.</li>
  <li><code>--overwrite &lt;T|F&gt;</code> &ndash; Replace existing files if true or fail when trying to overwrite a file if false. Default = <code>false</code>.</li>
  <li><code>--parameter-file &lt;string&gt;</code> &ndash; A file containing parameters.  See the <a href="parameter-file.html">parameter documentation</a> page for details. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--verbosity &lt;integer&gt;</code> &ndash; Specify the verbosity of the current processes. Each level prints the following messages, including all those at lower verbosity levels: 0-fatal errors, 10-non-fatal errors, 20-warnings, 30-information on the progress of execution, 40-more progress information, 50-debug info, 60-detailed debug info. Default = <code>30</code>.</li>
</ul>
</li>

</ul>
<hr>
<a href="/">Home</a>
</body>
</html>
