<!DOCTYPE HTML>
<html>
<head>
<meta charset="UTF-8">
<title>crux tide-index</title>
<script type="text/javascript"
  src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
</script>
<script type="text/javascript">
  MathJax.Hub.Config({jax: ["input/TeX","output/HTML-CSS"], displayAlign: "left"});
</script>
</head>
<body>
<h1>tide-index</h1>
<h2>Usage:</h2>
<p><code>crux tide-index [options] &lt;protein fasta file&gt; &lt;index name&gt;</code></p>
<h2>Description:</h2>
<p>Tide is a tool for identifying peptides from tandem mass spectra. It is an independent reimplementation of the SEQUEST<sup>&reg;</sup> algorithm, which assigns peptides to spectra by comparing the observed spectra to a catalog of theoretical spectra derived from a database of known proteins. Tide's primary advantage is its speed. Our published paper provides more detail on how Tide works. If you use Tide in your research, please cite:</p><blockquote>Benjamin J. Diament and William Stafford Noble. &quot;<a href="http://dx.doi.org/10.1021/pr101196n">Faster SEQUEST Searching for Peptide Identification from Tandem Mass Spectra.</a>&quot; <em>Journal of Proteome Research</em>. 10(9):3871-9, 2011.</blockquote><p>The <code>tide-index</code> command performs an optional pre-processing step on the protein database, converting it to a binary format suitable for input to the <code>tide-search</code> command.</p><p>Tide considers only the standard set of 20 amino acids. Peptides containing non-amino acid alphanumeric characters (BJOUXZ) are skipped. Non-alphanumeric characters are ignored completely.</p>
<h2>Input:</h2>
<ul>
  <li><code>&lt;protein fasta file&gt;</code> &ndash; The name of the file in FASTA format from which to retrieve proteins.</li>
  <li><code>&lt;index name&gt;</code> &ndash; The desired name of the binary index.</li>
</ul>
<h2>Output:</h2>
<p>The program writes files to the folder <code>crux-output</code> by default. The name of the output folder can be set by the user using the <code>--output-dir</code> option. The following files will be created:
<ul>
  <li><code>index</code> &ndash; A binary index, using the name specified on the command line.</li>
  <li><code>tide-index.params.txt</code> &ndash; a file containing the name and value of all parameters/options for the current operation. Not all parameters in the file may have been used in the operation. The resulting file can be used with the --parameter-file option for other crux programs.</li>
  <li><code>tide-index.log.txt</code> &ndash; a log file containing a copy of all messages that were printed to the screen during execution.</li>
</ul>
<h2>Options:</h2>
<ul style="list-style-type: none;">
<li>
<h3>Peptide properties</h3>
<ul>
  <li><code>--max-length &lt;integer&gt;</code> &ndash; The maximum length of peptides to consider. Default = <code>50</code>.</li>
  <li><code>--max-mass &lt;float&gt;</code> &ndash; The maximum mass (in Da) of peptides to consider. Default = <code>7200</code>.</li>
  <li><code>--min-length &lt;integer&gt;</code> &ndash; The minimum length of peptides to consider. Default = <code>6</code>.</li>
  <li><code>--min-mass &lt;float&gt;</code> &ndash; The minimum mass (in Da) of peptides to consider. Default = <code>200</code>.</li>
  <li><code>--monoisotopic-precursor &lt;T|F&gt;</code> &ndash; When computing the mass of a peptide, use monoisotopic masses rather than average masses. Default = <code>true</code>.</li>
  <li><code>--clip-nterm-methionine &lt;T|F&gt;</code> &ndash; When set to T, for each protein that begins with methionine, tide-index will put two copies of the leading peptide into the index, with and without the N-terminal methionine. Default = <code>false</code>.</li>
</ul>
</li>
<li>
<h3>Amino acid modifications</h3>
<ul>
  <li><code>--mods-spec &lt;string&gt;</code> &ndash; The general form of a modification specification has three components, as exemplified by <span style="color: red;">1</span><span style="color: green;">STY</span>+<span style="color: blue">79.966331</span>.<br>The three components are: [<span style="color: red;">max_per_peptide</span>]<span style="color: green;">residues</span>[+/-]<span style-"color: blue;">mass_change</span><br>In the example, <span style="color: red;">max_per_peptide</span> is <span style="color: red;">1</span>, <span style="color: green;">residues</span> are <span style="color: green;">STY</span>, and <span style="color: blue;">mass_change</span> is <span style="color: blue;">+79.966331</span>. To specify a static modification, the number preceding the amino acid must be omitted; i.e., <span style="color: green;">C</span>+<span style="color: blue;">57.02146</span> specifies a static modification of 57.02146 Da to cysteine. Note that Tide allows at most one modification per amino acid.  Also, the default modification (C+57.02146) will be added to every mods-spec string unless an explicit C+0 is included. Default = <code>C+57.02146</code>.</li>
  <li><code>--cterm-peptide-mods-spec &lt;string&gt;</code> &ndash; Specify peptide c-terminal modifications. See nterm-peptide-mods-spec for syntax. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--nterm-peptide-mods-spec &lt;string&gt;</code> &ndash; Specify peptide n-terminal modifications. Like --mods-spec, this specification has three components, but with a slightly different syntax. The <span style="color: red;">max_per_peptide</span> can be either "1", in which case it defines a variable terminal modification, or missing, in which case the modification is static. The <span style="color: green;">residues</span> field indicates which amino acids are subject to the modification, with the reside <span style="color: green;">X</span> corresponding to any amino acid. Finally, <span style="color: blue;">added_mass</span> is defined as before. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--max-mods &lt;integer&gt;</code> &ndash; The maximum number of modifications that can be applied to a single peptide. Default = <code>255</code>.</li>
  <li><code>--min-mods &lt;integer&gt;</code> &ndash; The minimum number of modifications that can be applied to a single peptide. Default = <code>0</code>.</li>
</ul>
</li>
<li>
<h3>Decoy database generation</h3>
<ul>
  <li><code>--decoy-format &lt;none|shuffle|peptide-reverse|protein-reverse&gt;</code> &ndash; Include a decoy version of every peptide by shuffling or reversing the target sequence or protein. In shuffle or peptide-reverse mode, each peptide is either reversed or shuffled, leaving the N-terminal and C-terminal amino acids in place. Note that peptides appear multiple times in the target database are only shuffled once. In peptide-reverse mode, palindromic peptides are shuffled. Also, if a shuffled peptide produces an overlap with the target or decoy database, then the peptide is re-shuffled up to 5 times. Note that, despite this repeated shuffling, homopolymers will appear in both the target and decoy database. The protein-reverse mode reverses the entire protein sequence, irrespective of the composite peptides. Default = <code>shuffle</code>.</li>
  <li><code>--keep-terminal-aminos &lt;N|C|NC|none&gt;</code> &ndash; When creating decoy peptides using decoy-format=shuffle or decoy-format=peptide-reverse, this option specifies whether the N-terminal and C-terminal amino acids are kept in place or allowed to be shuffled or reversed. For a target peptide "EAMPK" with decoy-format=peptide-reverse, setting keep-terminal-aminos to "NC" will yield "EPMAK"; setting it to "C" will yield "PMAEK"; setting it to "N" will yield "EKPMA"; and setting it to "none" will yield "KPMAE". Default = <code>NC</code>.</li>
  <li><code>--seed &lt;string&gt;</code> &ndash; When given a unsigned integer value seeds the random number generator with that value. When given the string "time" seeds the random number generator with the system time. Default = <code>1</code>.</li>
</ul>
</li>
<li>
<h3>Enzymatic digestion</h3>
<ul>
  <li><code>--enzyme &lt;no-enzyme|trypsin|trypsin/p|chymotrypsin|elastase|clostripain|cyanogen-bromide|iodosobenzoate|proline-endopeptidase|staph-protease|asp-n|lys-c|lys-n|arg-c|glu-c|pepsin-a|elastase-trypsin-chymotrypsin|custom-enzyme&gt;</code> &ndash; Specify the enzyme used to digest the proteins in silico. Available enzymes (with the corresponding digestion rules indicated in parentheses) include no-enzyme ([X]|[X]), trypsin ([RK]|{P}), trypsin/p ([RK]|[]), chymotrypsin ([FWYL]|{P}), elastase ([ALIV]|{P}), clostripain ([R]|[]), cyanogen-bromide ([M]|[]), iodosobenzoate ([W]|[]), proline-endopeptidase ([P]|[]), staph-protease ([E]|[]), asp-n ([]|[D]), lys-c ([K]|{P}), lys-n ([]|[K]), arg-c ([R]|{P}), glu-c ([DE]|{P}), pepsin-a ([FL]|{P}), elastase-trypsin-chymotrypsin ([ALIVKRWFY]|{P}). Specifying --enzyme no-enzyme yields a non-enzymatic digest. <strong>Warning:</strong> the resulting index may be quite large. Default = <code>trypsin</code>.</li>
  <li><code>--custom-enzyme &lt;string&gt;</code> &ndash; Specify rules for in silico digestion of protein sequences. Overrides the enzyme option. Two lists of residues are given enclosed in square brackets or curly braces and separated by a |. The first list contains residues required/prohibited before the cleavage site and the second list is residues after the cleavage site. If the residues are required for digestion, they are in square brackets, '[' and ']'. If the residues prevent digestion, then they are enclosed in curly braces, '{' and '}'. Use X to indicate all residues. For example, trypsin cuts after R or K but not before P which is represented as [RK]|{P}. AspN cuts after any residue but only before D which is represented as [X]|[D]. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--digestion &lt;full-digest|partial-digest|non-specific-digest&gt;</code> &ndash; Specify whether every peptide in the database must have two enzymatic termini (full-digest) or if peptides with only one enzymatic terminus are also included (partial-digest). Default = <code>full-digest</code>.</li>
  <li><code>--missed-cleavages &lt;integer&gt;</code> &ndash; Maximum number of missed cleavages per peptide to allow in enzymatic digestion. Default = <code>0</code>.</li>
</ul>
</li>
<li>
<h3>Input and output</h3>
<ul>
  <li><code>--decoy-prefix &lt;string&gt;</code> &ndash; Specifies the prefix of the protein names that indicate a decoy. Default = <code>decoy_</code>.</li>
  <li><code>--output-dir &lt;string&gt;</code> &ndash; The name of the directory where output files will be created. Default = <code>crux-output</code>.</li>
  <li><code>--overwrite &lt;T|F&gt;</code> &ndash; Replace existing files if true or fail when trying to overwrite a file if false. Default = <code>false</code>.</li>
  <li><code>--peptide-list &lt;T|F&gt;</code> &ndash; Create in the output directory a text file listing of all the peptides in the database, along with their neutral masses, one per line. If decoys are generated, then a second file will be created containing the decoy peptides. Decoys that also appear in the target database are marked with an asterisk in a third column. Default = <code>false</code>.</li>
  <li><code>--parameter-file &lt;string&gt;</code> &ndash; A file containing parameters.  See the <a href="parameter-file.html">parameter documentation</a> page for details. Default = <code>&lt;empty&gt;</code>.</li>
  <li><code>--verbosity &lt;integer&gt;</code> &ndash; Specify the verbosity of the current processes. Each level prints the following messages, including all those at lower verbosity levels: 0-fatal errors, 10-non-fatal errors, 20-warnings, 30-information on the progress of execution, 40-more progress information, 50-debug info, 60-detailed debug info. Default = <code>30</code>.</li>
</ul>
</li>

</ul>
<hr>
<a href="/">Home</a>
</body>
</html>
